optimus/lib/src/notification/notification_manager.dart

```diff
@@ -0,0 +1,329 @@
+import 'package:flutter/material.dart';
+import 'package:optimus/optimus.dart';
+
+/// The OptimusNotificationManager serves for handling OptimusNotifications.
+///
+/// By default maximum visible notifications is set to [_defaultMaxVisibleCount].
+/// You can change that by changing [customMaxVisibleCount].
+class OptimusNotificationManager {
+  factory OptimusNotificationManager() => _instance;
+
+  OptimusNotificationManager._();
+  static final OptimusNotificationManager _instance =
+      OptimusNotificationManager._();
+
+  final List<_NotificationModel> _visibleNotifications = [];
+  final List<_NotificationModel> _queuedNotifications = [];
+
+  int? _customMaxVisibleCount;
+
+  final GlobalKey<AnimatedListState> _listStateKey =
+      GlobalKey<AnimatedListState>();
+
+  OverlayEntry? _overlayEntry;
+
+  /// Notifications are served in the order they were created (First in, First out).
+  /// Each notification has self dismiss timer set to [_autoDismissDuration] and
+  /// is limited to a maximum width of [_maxWidth].
+  void showNotification({
+    required BuildContext context,
+    required String title,
+    String? body,
+    String? link,
+    IconData? icon,
+    VoidCallback? onLinkPressed,
+    VoidCallback? onDismissed,
+    OptimusNotificationVariant variant = OptimusNotificationVariant.info,
+  }) {
+    assert(
+      link == null && onLinkPressed == null ||
+          link != null && onLinkPressed != null,
+      'Can\'t provide link and have null onLinkPressed and vice versa',
+    );
+    assert(title.isNotEmpty, 'Can\'t create notification with empty title');
+    final _NotificationModel notification = _createNotificationModel(
+      title: title,
+      body: body,
+      link: link,
+      icon: icon,
+      onLinkPressed: onLinkPressed,
+      onDismissed: onDismissed,
+      variant: variant,
+    );
+    _processNotification(context, notification);
+  }
+
+  _NotificationModel _createNotificationModel({
+    required String title,
+    String? body,
+    String? link,
+    IconData? icon,
+    VoidCallback? onLinkPressed,
+    VoidCallback? onDismissed,
+    required OptimusNotificationVariant variant,
+  }) {
+    late _NotificationModel notification;
+    final onDismissPressed = onDismissed == null
+        ? null
+        : () {
+            _removeNotification(notification);
+          };
+    final onLinkPress = onLinkPressed == null
+        ? null
+        : () {
+            onLinkPressed.call();
```

Comment:
```suggestion
            onLinkPressed();
```

Same in other places.